File: csharp/MapleStory.OpenAPI/Src/Common/DTO/Character/CharacterStat.cs
namespace MapleStory.OpenAPI.Common.DTO
{
    public abstract class CharacterStatDTO<TCharacterFinalStat>
    {
        public abstract DateTimeOffset? Date { get; set; }
        public abstract string? CharacterClass { get; set; }
        public abstract List<TCharacterFinalStat> FinalStat { get; set; }
        public abstract int? RemainAp { get; set; }
    }

    public abstract class CharacterStatDTO : CharacterStatDTO<CharacterFinalStatDTO>
    {

    }

    public abstract class CharacterFinalStatDTO
    {
        public abstract string StatName { get; set; }
        public abstract string StatValue { get; set; }
    }
}

File: js/src/maplestory/api/common/dto/character/characterStat.ts
export abstract class CharacterStatDto {
  public abstract date: Date | null;
  public abstract characterClass: string;
  public abstract finalStat: CharacterFinalStatDto[];
  public abstract remainAP: number;
}

export abstract class CharacterFinalStatDto {
  public abstract statName: string;
  public abstract statValue: string;
}
